src/dot_cargo.rs

```diff
@@ -147,6 +148,10 @@ impl DotCargo {
         self.build = Some(DotCargoBuild::new(target));
     }
 
+    pub fn set_env(&mut self, env: &Option<toml::value::Table>) {
```

Comment:
I think I'd rather take this by value and have the "user" choose to clone at the call site.